app/controllers/challenge_assignments_controller.rb

```diff
@@ -67,21 +67,22 @@ def assignments_not_sent
     false
   end
 
-
-
   # ACTIONS
 
   def index
     if params[:user_id] && (@user = User.find_by(login: params[:user_id]))
       if current_user == @user
-        if params[:collection_id] && (@collection = Collection.find_by(name: params[:collection_id]))
-          @challenge_assignments = @user.offer_assignments.in_collection(@collection).undefaulted + @user.pinch_hit_assignments.in_collection(@collection).undefaulted
-        else
-          @challenge_assignments = @user.offer_assignments.undefaulted + @user.pinch_hit_assignments.undefaulted
-        end
+        @challenge_assignments = @user.assignments.undefaulted
+        @challenge_assignments = @challenge_assignments.in_collection(@collection) if params[:collection_id] && (@collection = Collection.find_by(name: params[:collection_id]))
+        @challenge_assignments = if params[:posted] == "true"
+                                   @challenge_assignments.posted
+                                 else
+                                   @challenge_assignments.unposted
+                                 end
+        @pagy, @challenge_assignments = pagy(@challenge_assignments.order(Arel.sql("COALESCE(sent_at, covered_at) ASC")))
```

Comment:
I think it's possible for multiple assignments to have the same `sent_at` or `covered_at`, especially if they're in the same challenge. We should probably add something more deterministic to the sort order as a fallback to avoid any issues with the pagination (e.g., the same item appears on two pages). Descending id would probably be fine.

Reply:
Updated!